File: src/lecture12/java/Main.java
package lecture12.java;

public class Main {

    public static void main(String[] args) {
        moveSomething(new Movable() {

            @Override
            public void move() {
                System.out.println("뚜벅 뚜벅");
            }

            @Override
            public void fly() {
                System.out.println("파닥 파닥");
            }
        });
    }

    private static void moveSomething(Movable movable) {
        movable.move();
        movable.fly();
    }
}

File: src/lecture12/kotlin/Main.kt
package lecture12.kotlin

import lecture12.java.Movable

fun main() {
    moveSomething(object : Movable {

        override fun move() {
            println("뚜벅 뚜벅")
        }

        override fun fly() {
            println("파닥 파닥")
        }
    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}
